src/org/rascalmpl/util/maven/Util.java

```diff
@@ -26,12 +26,41 @@
  */
 package org.rascalmpl.util.maven;
 
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
 import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.apache.maven.settings.Settings;
+import org.apache.maven.settings.io.xpp3.SettingsXpp3Reader;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
 
 /* package */ class Util {
     static Path mavenRepository() {
-        // TODO: parse settings.xml and figure out the repo
-        return Path.of(System.getProperty("user.home")).resolve(".m2").resolve("repository2");
+        String repoProp = System.getProperty("maven.repo.local");
+        if (repoProp != null) {
+            return Paths.get(repoProp);
+        }
+
+        SettingsXpp3Reader reader = new SettingsXpp3Reader();
+        Path userHome = Paths.get(System.getProperty("user.home"));
+        Path settingsXml = userHome.resolve(".m2").resolve("settings.xml");
```

Comment:
According to [maven](https://maven.apache.org/settings.html) there are 2 locations:

> There are two locations where a settings.xml file may live:
>  - The Maven install: ${maven.home}/conf/settings.xml
>  - A user's install: ${user.home}/.m2/settings.xml

We should make sure to support both of them (and figure out in what order)